engine/crates/engine-v2/src/plan/tracker.rs

```diff
@@ -0,0 +1,47 @@
+use super::PlanId;
+
+pub struct ExecutableTracker {
```

Comment:
Not really related to the PR. But pleased the borrow checker to separate mutable state from shared one and intended to it at some point anyway.